### PR TITLE
Add MerkleMountainRangeRoot digest item

### PR DESCRIPTION
**Pull request by HackFisher**
This is to add a new type of digest item in digest, 
```
MerkleMountainRangeRoot(Hash)
```
This is used for future light-client enhancement, and here is a sample pallet how this digest log will be deposited:

https://github.com/darwinia-network/darwinia-common/blob/v0.5.0/frame/header-mmr/src/lib.rs#L85

Related https://github.com/darwinia-network/substrate/pull/1 and https://github.com/darwinia-network/darwinia/pull/358

DigestItemType in darwinia's fork was using 18 to avoid future conflicts with the potential reserved types, this PR is going to use 8 as the type index, and will sync darwinia's forking to the same 8 after this PR get merged.

Thanks.

### CODE REVIEW

**Comment by HackFisher** [comment]
[clabot:check].

**Comment by NikVolf** [comment]
Can't you use `DigestItem::Other`?

Also `Hash` generic paramter is used for trie root digest item

Can we be sure that MMR root hash is even related to this and be the same?

File: primitives/runtime/src/generic/digest.rs [hunk 1]
```diff
@@ -173,6 +178,10 @@ pub enum DigestItemRef<'a, Hash: 'a> {
 	/// Digest item that contains signal from changes tries manager to the
 	/// native code.
 	ChangesTrieSignal(&'a ChangesTrieSignal),
+
+	/// Reference to `DigestItem::MerkleMountainRangeRoot`.
+	MerkleMountainRangeRoot(&'a Hash),
```
Comment (tomusdrw) [inline primitives/runtime/src/generic/digest.rs:183]: Why isn't this done via `Other` case? You can create your own extension trait in your own code.
To handle the conversion like so:

```rust
trait AsMerkleMountainRangeRoot<Hash> {
 fn as_merkle_mountain_range_root(&self) -> Option<&Hash>;
}

impl<'a, Hash: 'a> AsMerkleMountainRangeRoot<Hash> for DigestItemRef<'a, Hash> {
  fn as_merkle_mountain_range_root(&self) -> Option<&Hash> {
     match *self {
			DigestItemRef::Other(ref data) => Hash::from(&data).ok(),
			_ => None,
		}
  }
}
```

Comment (HackFisher) [reply primitives/runtime/src/generic/digest.rs:183]: It makes sense.

I was thinking that if we use ```Other```, the data may conflict with digest item having the same struct which could be introduced in future modules. 

I think can resolve this by using a specific prefix in ```Other``` data.

Thanks, I'm closing this PR.

Comment (tomusdrw) [reply primitives/runtime/src/generic/digest.rs:183]: AFAICT `Other` is meant to be exactly this - a non-conflicting way to add your own thing to `Digest`. Unless someone corrects me ;)
It is indeed to only one element though, but internally you can then encode multiple different variants.
Perhaps we are lacking some documentation here to make it clear what's the intention of that variant.

**Pull request closed without merging**